api/python/quilt3/util.py

```diff
@@ -140,7 +144,9 @@ def __init__(self, bucket, path, version_id):
             if os.name == 'nt':
                 assert '\\' not in path, "Paths must use / as a separator"
         else:
-            assert not path.startswith('/'), "S3 paths must not start with '/'"
+            if path.startswith('/'):
+                # S3 paths are not absolute, but there may be a "/" folder in root.
```

Comment:
this comment makes my brain hurt a lot because of leaky abstractions (S3 don't really have paths, it has object keys)